libindy/src/commands/anoncreds/prover.rs

```diff
@@ -279,7 +279,7 @@ impl ProverCommandExecutor {
 
         let credential_request_metadata = CredentialRequestMetadata {
             master_secret_blinding_data: ms_blinding_data,
-            nonce: credential_request.nonce.clone()?,
+            nonce: credential_request.nonce.try_clone()?,
```

Comment:
leaving a note for myself to compare the differences between **clone()** and **try_clone()**

Reply:
From what I can tell, this **try_clone()** function is performing a different operation than the **clone()** trait which was originally used. **try_clone()** appears to be generating a new BigNum. I got to this by looking at this line here: https://github.com/hyperledger/ursa/blob/0d03b1f071af652e3fa849241546dd0867ad3d51/libursa/src/bn/openssl.rs#L404

which then got me to look at the BigNum impl documentation in SSL, which says "Creates a new `BigNum` from an unsigned, big-endian encoded number of arbitrary length."

https://docs.rs/crate/openssl/0.10.20/source/src/bn.rs

Should we be generating a new nonce for this or do we want to be cloning an already created nonce?

Reply:
No issues here. **clone()** is not a reference to a trait, but rather to an indy_crypto function found here. This is just a function rename to create more idiomatic rust. See the original function called here: https://github.com/hyperledger/indy-crypto/blob/1675e29a2a5949b44899553d3d128335cf7a61b3/libindy-crypto/src/bn/openssl.rs#L404